### PR TITLE
Fix copy and pasting a folder recursively.

### PR DESCRIPTION
**Pull request by kumudhakn**
Signed-off-by: kumudhakn <kumudha.n@samsung.com>

### CODE REVIEW

File: packages/filesystem/src/browser/file-tree/file-tree-model.ts [hunk 1]
```diff
@@ -148,6 +148,13 @@ export class FileTreeModel extends TreeModelImpl implements LocationService {
             return false;
         }
         const targetUri = node.uri.resolve(uri.path.base);
+        /* Check if the folder is copied on itself */
+        const sourcePath = uri.path.toString();
+        const targetPath = node.uri.path.toString();
+        if (sourcePath === targetPath) {
+            return true;
```
Comment (akosyakov) [inline packages/filesystem/src/browser/file-tree/file-tree-model.ts]: `return false`

**Comment by kumudhakn** [comment]
I am new to Theia and I am not very sure how to differentiate between a directory and simlink from the code. Can you give me some pointers for the same?

Thanks.

**Comment by marechal-p** [comment]
Hi @kumudhakn, I think we can forget about the symlinks for now, as it seems like we would need to add a bunch of new APIs in order to properly support it. Until then, I feel like your patch does cover the most common corner case :)

**Comment by marechal-p** [comment]
@kumudhakn Please just squash your changes to have only 1 commit, and make sure that it is signed-off :)

**Comment by kumudhakn** [comment]
@marechal-p The commits have been sqaushed. Do let me know if you need anything else. Thanks!

**Pull request merged**